src/cheevos/Cheevos.cpp

```diff
@@ -0,0 +1,285 @@
+/*
+ *  This file is based on https://github.com/libretro/RetroArch/blob/96c5f5dfb07454c972c50838815330382d6b1911/cheevos/fixup.c
+ *
+ *  Copyright (C) 2014-2020 Team Kodi (https://kodi.tv)
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSE.md for more information.
+ */
+
+#include "Cheevos.h"
+
+#include "libretro/LibretroEnvironment.h"
+#include "libretro/MemoryMap.h"
+#include "libretro/libretro.h"
+#include "rcheevos/rconsoles.h"
+#include "rcheevos/rhash.h"
+#include "rcheevos/rurl.h"
+
+using namespace LIBRETRO;
+
+namespace
+{
+constexpr int HASH_SIZE = 33;
+constexpr int URL_SIZE = 512;
+constexpr int RICH_PRESENCE_EVAL_SIZE = 512;
+constexpr int POST_DATA_SIZE = 1024;
+} // namespace
+
+CCheevos::CCheevos()
+{
+  rc_runtime_init(&m_runtime);
+}
+
+void CCheevos::Initialize()
+{
+  rc_runtime_init(&m_runtime);
+}
+
+void CCheevos::Deinitialize()
+{
+  rc_runtime_destroy(&m_runtime);
+}
+
+void CCheevos::ResetRuntime()
+{
+  rc_runtime_reset(&m_runtime);
+  m_richPresence =
+      rc_parse_richpresence(m_richPresenceBuffer.data(), m_richPresenceScript.c_str(), NULL, 0);
+}
+
+CCheevos& CCheevos::Get(void)
+{
+  static CCheevos _instance;
+  return _instance;
+}
+
+bool CCheevos::GenerateHashFromFile(std::string& hash,
+                                    unsigned int consoleID,
+                                    const std::string& filePath)
+{
+  char _hash[HASH_SIZE] = {};
+
+  int res = rc_hash_generate_from_file(_hash, consoleID, filePath.c_str());
+  hash = _hash;
+
+  return res != 0;
```

Comment:
Instead of `0` use the value defined in `rc_hash_generate_from_file()`'s documentation: `RC_OK`.